src/osd/PrimaryLogPG.cc

```diff
@@ -1790,11 +1788,9 @@ void PrimaryLogPG::do_request(
     op->pg_trace.init("pg op", &trace_endpoint, &op->osd_trace);
     op->pg_trace.event("do request");
   }
-#ifdef HAVE_JAEGER
   if (op->osd_parent_span) {
-    auto do_req_span = jaeger_tracing::child_span(__func__, op->osd_parent_span);
+    tracer.start_span(__func__, op->osd_parent_span);
```

Comment:
please remove the "if" statement, passing null as a parent span should be ok
please cache to a local variable - so we can finish in the dtor at the end of the function

Reply:
@ideepika 
hey, 
after that fix, `make check` has failed on 
`../src/osd/ReplicatedBackend.cc:1061:8: error: unused variable 'span' [-Werror,-Wunused-variable]
  auto span = tracer.start_span(__func__, op->osd_parent_span);`
  
is there something we can do about it except using the variable?

Reply:
@zenomri how about using:
`
[[maybe_unused]] span = tracer.start_span(__func__, op->osd_parent_span);`